site/app/libraries/database/DatabaseQueries.php

```diff
@@ -144,16 +144,37 @@ public function updateSubmittyUserTimeZone(User $user, string $time_zone) {
 
     /**
      * Gets a user from the database given a user_id.
+     *
+     * @param string $user_id
+     * @return User|null
      */
     public function getUserById(string $user_id): ?User {
         return $this->getUser($user_id);
     }
 
-    public function getUserByNumericId($numeric_id): ?User {
+    /**
+     * Gets a list of first and last names from the database given an array of IDs.
+     *
+     * @param array $user_id_list
+     * @return array|null
+     */
+    public function getUsersByIds(array $user_id_list): ?array {
+        return $this->getUsers($user_id_list);
+    }
+
+    /**
+     * @param int $numeric_id
+     * @return User|null
+     */
+    public function getUserByNumericId(int $numeric_id): ?User {
         return $this->getUser($numeric_id, true);
     }
 
-    public function getUserByIdOrNumericId($id) {
+    /**
+     * @param int|string $id
+     * @return User|null
+     */
+    public function getUserByIdOrNumericId($id): ?User {
```

Comment:
I know you didn't pick this function name but does it make sense to keep it like this. This calls getUser with $id and no is_numeric field which means it will always be by normal id and never numeric id.

Reply:
I was a little confused about the purpose of this function as well.  If I'm not mistaken, this function does get a user by either the normal user ID or the numeric ID.  The function essentially just runs the query again with the is_numeric flag if the regular ID couldn't be found.  I believe the only place this function is called is here: https://github.com/Submitty/Submitty/blob/1704fd8fd4a82010d8a8d904c35b8b2d28be6f31/site/app/views/submission/HomeworkView.php#L637

Perhaps the best thing to do here would be to just rewrite the place where this function is called and delete the function altogether.  It would be better to use `is_numeric()` to make the determination about what type of ID it is instead of just running the query twice...

Reply:
I think I misread the code first initially. It definitely works but yeah I agree that there really isn't a need for it to have 2 queries when you can quickly check if it's numeric without the database.

Reply:
I'll make the changes described above and remove the function.